queries.sql

```diff
@@ -1,5 +1,111 @@
 /*Queries that provide answers to the questions from all projects.*/
 
+/* FEATURE BRANCH 04-add-join-table-for-visits */
+
+--D) Write queries to answer thhe following...
+
+-- D1) Who was the last animal seen by William Tatcher?
+SELECT animals.name as animal, vets.name as vet, 
+visits.date_of_visit
+FROM visits
+JOIN animals ON (visits.animal_id = animals.id) 
+JOIN vets ON (visits.vet_id = vets.id 
+	AND vets.name = 'William Tatcher')
+WHERE (visits.date_of_visit = (SELECT MAX(visits.date_of_visit) 
+	FROM visits 
+		JOIN animals ON (visits.animal_id = animals.id) 
+		JOIN vets ON (visits.vet_id = vets.id 
+			AND vets.name = 'William Tatcher')));
+
+-- D2) How many different animals did Stephanie Mendez see?
+SELECT animals.name as animal, vets.name as vet 
+FROM visits
+JOIN animals ON (visits.animal_id = animals.id) 
+JOIN vets ON (visits.vet_id = vets.id 
+	AND vets.name = 'Stephanie Mendez')
+GROUP BY animals.name, vets.name;
+
+-- D3) List all vets and their specialties, including vets with
+--     no specialties:
+SELECT v.name, s.species_id, sp.name as specialty
+FROM vets as v
+LEFT JOIN specializations as s ON (v.id = s.vet_id)
+LEFT JOIN species as sp ON (s.species_id = sp.id);
+
+-- D4) List all animals that visited Stephanie Mendez
+--     between April 1st and August 30th, 2020
+SELECT a.name as animal, ve.name as vet, 
+v.date_of_visit
+FROM visits as v
+JOIN animals as a ON (v.animal_id = a.id)
+JOIN vets as ve ON (v.vet_id = ve.id)
+ORDER BY (a.name, v.date_of_visit);
+
+-- D5) What animal has the most visits to vets?
+SELECT a.name as animal, ve.name as vet, 
+COUNT (v.date_of_visit) as visits
+FROM visits as v
+JOIN animals as a ON (v.animal_id = a.id)
+JOIN vets as ve ON (v.vet_id = ve.id)
+GROUP BY (a.name, ve.name)
+ORDER BY visits DESC;
```

Comment:
- [ ] Hello! 🌟 It might be a great idea to apply a limit to the search results, ensuring that it returns just one row. 📏🔍 Let's explore this option for a more concise and specific outcome. This can be helpful when we want to highlight a single, most relevant piece of information, making the result more focused and easier to digest. 🧐📌